server/application-server/src/main/java/de/tum/cit/aet/helios/deployment/DeploymentController.java

```diff
@@ -50,13 +51,13 @@ public ResponseEntity<DeploymentDto> getLatestDeploymentByEnvironmentId(
   }
 
   @PostMapping("/deploy")
-  public ResponseEntity<String> deployToEnvironment(@RequestBody DeployRequest deployRequest) {
+  public ResponseEntity<String> deployToEnvironment(
+      @RequestHeader("Authorization") String token, @RequestBody DeployRequest deployRequest) {
     try {
-      deploymentService.deployToEnvironment(deployRequest);
+      deploymentService.deployToEnvironment(deployRequest, token);
```

Comment:
issue: please check if the token is set before calling the deploy function (not sure if this is an authenticated route though) and return an unauthenticated error instead of the internal server error